### PR TITLE
Course Unit Enrollments now have their state defined in an enum-like …

### PR DESCRIPTION
**Pull request by mehalshah**
…thing (in case we have to add more).

Also, change the teacher view so we just display the status of the course unit instead of the status of the task assignments in there, bringing this more in line with the spec

### CODE REVIEW

File: dashboard/app/models/plc/enrollment_unit_assignment.rb [hunk 1]
```diff
@@ -23,6 +23,15 @@ class Plc::EnrollmentUnitAssignment < ActiveRecord::Base
   has_many :plc_module_assignments, class_name: '::Plc::EnrollmentModuleAssignment', foreign_key: 'plc_enrollment_unit_assignment_id', dependent: :destroy
   has_many :plc_task_assignments, through: :plc_module_assignments, class_name: '::Plc::EnrollmentTaskAssignment', dependent: :destroy
 
+  START_BLOCKED = 'start_blocked'
+  PENDING_EVALUATION = 'pending_evaluation'
+  IN_PROGRESS = 'in_progress'
+  COMPLETED = 'completed'
+
+  unit_status_states = [START_BLOCKED, PENDING_EVALUATION, IN_PROGRESS, COMPLETED]
```
Comment (aoby) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb]: You can also do something like

``` ruby
UNIT_STATUS_STATES = [
  START_BLOCKED = 'start_blocked'
  PENDING_EVALUATION = 'pending_evaluation'
  IN_PROGRESS = 'in_progress'
  COMPLETED = 'completed'
]
```

which will create all the constants and the container array at once.

Comment (mehalshah) [inline dashboard/app/models/plc/enrollment_unit_assignment.rb]: Oh I like that better.

File: dashboard/test/models/plc/enrollment_task_assignment_test.rb [hunk 2]
```diff
@@ -24,16 +25,16 @@ class Plc::EnrollmentTaskAssignmentTest < ActiveSupport::TestCase
     task_assignments.first.complete_assignment!
     assert_equal 'completed', task_assignments.first.status
     unit_enrollment.reload
-    assert_not_equal 'completed ', unit_enrollment.status
+    assert_equal Plc::EnrollmentUnitAssignment::IN_PROGRESS, unit_enrollment.status
```
Comment (aoby) [inline dashboard/test/models/plc/enrollment_task_assignment_test.rb:30]: nit: just to make sure the state is as expected after setup, assert status == START_BLOCKED before calling complete_assignment! Same below.

**Comment by aoby** [comment]
lgtm

**Pull request merged**